src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,8 +32,24 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } else {
+            breakUpAddress(address);
         }
-        this.value = trimmedAddress;
+        this.value = block.getBlock() + COMMA + street.getStreet() + COMMA + unit.getUnit() + COMMA + postalCode.getPostalCode();
+    }
+
+    /**
+     * Breaking up the address into its attributes: Block, Street, Unit and PostalCode
+     * @param address
+     */
+    private void breakUpAddress(String address) {
+        int firstCommaIndex = address.indexOf(COMMA);
+        block = new Block(address.substring(0,firstCommaIndex));
```

Comment:
You should qualify direct access to the member variables by using the 'this' keyword. It makes it explicit and clear to the reader.
However, in this case you should be using getters and setters as discussed in class.